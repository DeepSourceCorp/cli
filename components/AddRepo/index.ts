import AddRepoModal from './AddRepoModal.vue'
import RepoCard from './RepoCard.vue'
import SyncRepoAlert from './SyncRepoAlert.vue'

export { AddRepoModal, RepoCard, SyncRepoAlert }
