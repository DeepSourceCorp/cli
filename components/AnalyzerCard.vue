<template>
  <div class="relative h-72 min-h-1/2 md:h-40 bg-ink-400 p-2 rounded-sm flex flex-col space-y-3 md:min-w-1/2 md:w-1/2 box-border overflow-scroll"
        :class="{ 'justify-center items-center space-y-2': configList.length }">
    <!-- Header -->
    <div v-if="configList.length > 0" class="flex w-full items-center space-x-2">
        <z-icon :icon="icon" size="small" color="robin"></z-icon>
        <!-- TODO: Get Repo name and handle name -->
        <div class="flex-1 text-sm text-vanilla-200 font-bold">
            {{ name }}
        </div>
        <div class="cursor-pointer" @click="handleClick">
            <z-icon icon="x"
                size="small" 
                color="vanilla-200">
            </z-icon>
        </div>
    </div>
    <!-- Config Section -->
    <template v-for="config in configList">
        <div :key="config.title" class="flex flex-col space-y-2">
            <div class="text-xs tracking-wide text-slate uppercase">
                {{config.title}}
            </div>
            <div v-if="config.type == 'boolean'">
                <z-radio-group v-model="config.selected" class="flex space-x-4 text-xs">
                    <z-radio v-for="option in config.values" 
                            :key="option.value"
                            :value="option.value"
                            :label="option.label"
                            ></z-radio>
                </z-radio-group>
            </div>
            <div v-else-if="config.type == 'input'">
                <z-input v-model="config.selected" 
                        textSize="xs">
                </z-input>
            </div>
            <div v-else-if="config.type == 'list'" class="h-8">
                <!-- TODO: Customizable placeholder -->
                <z-select 
                    v-model="config.selected" 
                    class="z-10" 
                    placeholder="Choose a framework">
                    <z-option
                        v-for="item in config.values"
                        :key="item.value"
                        :label="item.label"
                        :value="item.value">
                        <div class="flex items-center">
                            <z-icon icon="activity" size="small" class="-mt-1 mr-2"></z-icon>
                            <span class="flex-1">{{item.label}}</span>
                        </div>
                    </z-option>
                </z-select>
            </div>
        </div>
    </template>
    <!-- No Config Data Section -->
    <template v-if="configList.length == 0">
        <div class="absolute top-2 right-2 cursor-pointer" @click="handleClick">
            <z-icon icon="x"
                size="small" 
                color="vanilla-200">
            </z-icon>
        </div>
        <z-icon :icon="icon" color="pink"></z-icon>
        <div class="font-bold text-vanilla-100 text-sm">{{name}}</div>
    </template>
  </div>
</template>

<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator';
import { ZIcon, ZInput, ZSelect, ZOption, ZRadioGroup, ZRadio } from '@deepsourcelabs/zeal';
@Component({
    components: {
       ZIcon,
       ZInput,
       ZSelect,
       ZOption,
       ZRadioGroup,
       ZRadio
    }
})
export default class AnalyzerCard extends Vue {
    @Prop({default: 'lock'})
    icon!: string
    @Prop({default: ''})
    name!: string
    @Prop({default: []})
    configList!: Array<Record<string, unknown>>
    public handleClick(event: Event): void {
        this.$emit('click', event)
    }
}
</script>
