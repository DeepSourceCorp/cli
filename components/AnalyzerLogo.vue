<template>
  <div>
    <img
      v-if="analyzerLogo"
      v-tooltip="hideTooltip ? '' : name"
      class="w-auto flex-shrink-0"
      :class="sizeStyle"
      :src="analyzerLogo"
      :alt="name"
    />
    <z-icon
      v-else
      v-tooltip="hideTooltip ? '' : name"
      :icon="shortcode"
      color="transparent"
      :size="size"
    />
  </div>
</template>
<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator'

@Component({})
export default class AnalyzerLogo extends Vue {
  @Prop({ default: '' })
  analyzerLogo!: string

  @Prop({ default: '' })
  name!: string

  @Prop({ default: '' })
  shortcode!: string

  @Prop({ default: 'base' })
  size!: string

  @Prop({ default: false })
  hideTooltip: boolean

  get sizeStyle() {
    const sizes: Record<string, string> = {
      'x-small': 'w-3 h-3',
      small: 'w-4 h-4',
      base: 'w-5 h-5',
      medium: 'w-6 h-6',
      large: 'w-8 h-8'
    }

    return sizes[this.size] || 'w-5 h-4'
  }
}
</script>
