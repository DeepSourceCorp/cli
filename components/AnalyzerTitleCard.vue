<template>
  <div class="px-2 py-2 bg-ink-100 rounded-sm bg-ink-400 flex items-center space-x-2 cursor-pointer"
        @click="handleClick()">
    <z-icon :icon="icon" size="small" color="honey"></z-icon>
    <div class="flex-1 text-sm text-vanilla-200 font-bold">{{name}}</div>
    <z-icon icon="plus" size="small" class="cursor-pointer" color="vanilla-200"></z-icon>
  </div>
</template>

<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator';
import { ZIcon } from '@deepsourcelabs/zeal';
@Component({
    components: {
       ZIcon 
    }
})
export default class AnalyzerTitleCard extends Vue {
    @Prop({default: 'globe'})
    icon!: string
    @Prop({default: ''})
    name!: string
    public handleClick(event: Event): void {
        this.$emit('click', event)
    }
}
</script>
