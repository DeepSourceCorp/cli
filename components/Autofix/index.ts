import InfoIcon from './InfoIcon.vue'
import Info from './Info.vue'
import AutofixCard from './AutofixCard.vue'
import AutofixListItem from './AutofixListItem.vue'
import AutofixCodeDiff from './AutofixCodeDiff.vue'

export { InfoIcon, AutofixCard, AutofixListItem, AutofixCodeDiff, Info }
