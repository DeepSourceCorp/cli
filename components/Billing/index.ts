import PlanInfo from './PlanInfo.vue'
import BillingInfo from './BillingInfo.vue'
import InvoiceList from './InvoiceList.vue'
import UpdateBillingDetailsModal from './Modals/UpdateBillingDetailsModal.vue'

export { PlanInfo, BillingInfo, InvoiceList, UpdateBillingDetailsModal }
