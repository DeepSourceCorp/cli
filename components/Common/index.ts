import SidebarMenu from './SidebarMenu.vue'

export { SidebarMenu }
