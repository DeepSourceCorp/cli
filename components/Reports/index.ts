import DateRangePicker from './DateRangePicker.vue'
import RecentStats from './RecentStats.vue'
import OccurrenceTags from './OccurrenceTags.vue'
import ChartContainer from './ChartContainer.vue'
import ChartStat from './ChartStat.vue'

export { DateRangePicker, RecentStats, OccurrenceTags, ChartContainer, ChartStat }
