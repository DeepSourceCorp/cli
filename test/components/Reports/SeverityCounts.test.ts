import { render } from '@testing-library/vue'
import { SeverityCounts } from '~/components/Reports'

test('renders SeverityCounts with all prop options', () => {
  const props = {
    high: 1,
    low: 2,
    medium: 0
  }

  const { html } = render(SeverityCounts, {
    props,
    stubs: {
      ZIcon: true
    }
  })

  expect(html()).toMatchSnapshot(JSON.stringify(props))
})
