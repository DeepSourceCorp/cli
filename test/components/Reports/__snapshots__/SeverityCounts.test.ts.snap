// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SeverityCounts with all prop options: {"high":1,"low":2,"medium":0} 1`] = `
"<div class=\\"flex gap-x-12 text-xs font-semibold\\">
  <chartstat-stub title=\\"critical\\">
    <div class=\\"flex items-center gap-x-2\\">
      <zicon-stub icon=\\"issue-critical\\" color=\\"cherry\\" size=\\"x-small\\"></zicon-stub> <span class=\\"text-xs font-semibold\\">1</span>
    </div>
  </chartstat-stub>
  <chartstat-stub title=\\"major\\">
    <div class=\\"flex items-center gap-x-2\\">
      <zicon-stub icon=\\"issue-major\\" color=\\"honey\\" size=\\"x-small\\"></zicon-stub> <span class=\\"text-xs font-semibold\\">0</span>
    </div>
  </chartstat-stub>
  <chartstat-stub title=\\"minor\\">
    <div class=\\"flex items-center gap-x-2\\">
      <zicon-stub icon=\\"issue-minor\\" color=\\"vanilla-400\\" size=\\"x-small\\"></zicon-stub> <span class=\\"text-xs font-semibold\\">2</span>
    </div>
  </chartstat-stub>
</div>"
`;
